ops/testing.py

```diff
@@ -191,12 +188,32 @@ def add_oci_resource(self, resource_name: str,
             raise RuntimeError('Resource {} is not a defined resources'.format(resource_name))
         if self._meta.resources[resource_name].type != "oci-image":
             raise RuntimeError('Resource {} is not an OCI Image'.format(resource_name))
-        resource_dir = self._resource_dir / resource_name
-        resource_dir.mkdir(exist_ok=True)
-        resource_file = resource_dir / "contents.yaml"
-        with resource_file.open('wt', encoding='utf8') as resource_yaml:
-            yaml.dump(contents, resource_yaml)
-        self._backend._resources_map[resource_name] = resource_file
+
+        as_yaml = yaml.dump(contents, Dumper=yaml.SafeDumper)
+        self._backend._resources_map[resource_name] = ('contents.yaml', as_yaml)
+
+    def add_resource(self, resource_name: str, content: typing.AnyStr) -> None:
+        """Add content for a resource to the backend.
+
+        This will register the content, so that a call to `Model.resources.fetch(resource_name)`
+        will return a path to a file containing that content.
+
+        Args:
+            resource_name: The name of the resource being added
+            contents: Either string or bytes content, which will be the content of the filename
+                returned by resource-get. If contents is a string, it will be encoded in utf-8
```

Comment:
> If contents is a string, it will be encoded in utf-8

I'm assuming this is intended to mean the same as 'it should be encoded in utf-8', but I'm not 100% sure that's you mean, if you know what I mean, you know?

Reply:
If it is a string, it will be utf-8 encoded into the file contents.
If you have better wording, I'm happy to change it.